DentOS_Framework/DentOsTestbed/src/dent_os_testbed/test/test_suite/functional/bridging/test_bridging_static_entries.py

```diff
@@ -0,0 +1,140 @@
+    assert out[0][device_host_name]["rc"] == 0, err_msg
+
+    out = await IpLink.set(
+        input_data=[{device_host_name: [
+            {"device": port, "master": bridge, "operstate": "up"} for port in ports]}])
+    err_msg = f"Verify that bridge entities set to 'UP' state and links enslaved to bridge.\n{out}"
+    assert out[0][device_host_name]["rc"] == 0, err_msg
+
+    out = await BridgeLink.set(
+        input_data=[{device_host_name: [
+            {"device": port, "learning": False, "flood": False} for port in ports]}])
+    err_msg = f"Verify that entities set to learning 'OFF' and flooding 'OFF' state.\n{out}"
+    assert out[0][device_host_name]["rc"] == 0, err_msg
+
+    out = await BridgeFdb.add(
+        input_data=[{device_host_name: [
+            {"device": ports[0], "lladdr": "aa:bb:cc:dd:ee:11", "master": True, "static": True},
+            {"device": ports[1], "lladdr": "aa:bb:cc:dd:ee:12", "master": True, "static": True},
+            {"device": ports[2], "lladdr": "aa:bb:cc:dd:ee:13", "master": True, "static": True},
+            {"device": ports[3], "lladdr": "aa:bb:cc:dd:ee:14", "master": True, "static": True},
+            ]}])
+    assert out[0][device_host_name]["rc"] == 0, f"Verify that FDB static entries added.\n{out}"
+
+    address_map = (
+        #swp port, tg port,     tg ip,     gw,        plen
+        (ports[0], tg_ports[0], "1.1.1.2", "1.1.1.1", 24),
+        (ports[1], tg_ports[1], "2.2.2.2", "2.2.2.1", 24),
+        (ports[2], tg_ports[2], "3.3.3.2", "3.3.3.1", 24),
+        (ports[3], tg_ports[3], "4.4.4.2", "4.4.4.1", 24),
+    )
+
+    dev_groups = tgen_utils_dev_groups_from_config(
+        {"ixp": port, "ip": ip, "gw": gw, "plen": plen}
+        for _, port, ip, gw, plen in address_map
+    )
+
+    await tgen_utils_traffic_generator_connect(tgen_dev, tg_ports, ports, dev_groups)
+    
+    list_macs = ["aa:bb:cc:dd:ee:11", "aa:bb:cc:dd:ee:12",
+                 "aa:bb:cc:dd:ee:13", "aa:bb:cc:dd:ee:14"]
+
+    streams = {
+        f"bridge_{dst + 1}": {
+            "ip_source": dev_groups[tg_ports[src]][0]["name"],
+            "ip_destination": dev_groups[tg_ports[dst]][0]["name"],
+            "srcMac": list_macs[src],
+            "dstMac": list_macs[dst],
+            "type": "raw",
+            "protocol": "802.1Q",
+        } for src, dst in ((3, 0), (2, 1), (1, 2), (0, 3))
+    }
+
+    await tgen_utils_setup_streams(tgen_dev, config_file_name=None, streams=streams)
+
+    await tgen_utils_start_traffic(tgen_dev)
+    await asyncio.sleep(traffic_duration)
+    await tgen_utils_stop_traffic(tgen_dev)
+
+    # check the traffic stats
+    stats = await tgen_utils_get_traffic_stats(tgen_dev, "Traffic Item Statistics")
+    for row in stats.Rows:
+        assert float(row["Loss %"]) == 0.000, f'Failed>Loss percent: {row["Loss %"]}'
+        assert tgen_utils_get_loss(row) == 0.000, \
+        f"Verify that traffic from {row['Tx Port']} to {row['Rx Port']} forwarded.\n{out}"
```

Comment:
Just wanted to ensure, do we know if the traffic was received entirely from the right port with relevant dst mac?

Reply:
Yes, we verify that the traffic has been sent from each port to their respective ports.